### ISSUE TITLE
Gestures not working

### ISSUE THREAD

**Issue by peter099** [florisboard/florisboard#1397]
'Glide typing' and 'General gestures' are not working for me. I installed Florisboard from fdroid and i have LineageOS installed on my phone moto x play.

**Comment by Glitchy-Tozier** [comment]
Hi, and thank you for taking the time to write this bug-report! Two things:
1. Glide typing and most normal gestures can't coexist. I assume you could see that though.
2. Could you please fill in the full bug-template? It's really important to have as much information as possible, to be able to fix your bug.

**Comment by peter099** [comment]
Yes, i know glide typing and normal gestures can't coexist. If i enable glide typing, normal gestures are disabled automatically. 

I did more testing and i can say that glide typing is not working at all. But normal gestures seems to be working, but responsiveness is very weak. I have to swipe my finger very fast otherwise the swipe won't be detected. My current settings for normal gestures are: swipe up: shift, swipe down: hide keyboard, swipe left: delete current word, swipe right: insert space. 

Additional info:
- FlorisBoard Version: 0.3.13
- Install Source: F-Droid
- Device: Motorola X Play
- Android: LeneageOS 17

**Comment by Glitchy-Tozier** [comment]
Thank you for adding the info, let's see what Patrick can do when he gets around to reading this.
(I'm thinking this may already have been fixed and will be gone in the next version, but I'm not 100% sure)

**Comment by patrickgold** [comment]
Thanks for your bug report!

>  (I'm thinking this may already have been fixed and will be gone in the next version, but I'm not 100% sure)

Yes and no - the current builds allow for the gesture thresholds to be set directly (no more very fast/fast/normal/slow/very slow), thus allowing for a much finer control over the detection of gestures. Additionally the gesture threshold min/max have been adjusted , so you can go beyond limits of earlier versions.

> I did more testing and i can say that glide typing is not working at all.

This is something I do not understand. The next beta06 will probably lack support for glide typing though (because the whole keyboard UI backend is completely rewritten and I haven't gotten to implement glide typing again), so I think in beta07 we can try out if the re-implementation of glide typing makes thinks better or worse for you.

**Comment by peter099** [comment]
Thanks for your work. I hope the gestures are improved in the next releases.

**Comment by bittner** [comment]
I'm a [Volla Phone](https://volla.online/) user and glide typing works for me for the English keyboard (and only for English as [mentioned in the README](https://github.com/florisboard/florisboard#highlighted-features)).

FlorisBoard seems to be [officially endorsed](https://forum.volla.online/viewtopic.php?f=4&t=104) by the Volla Phone project. It would be great to see glide typing working for all languages soon again. And yes, I like the polished aesthetics of FlorisBoard. :heart_eyes:

**Comment by scrollop** [comment]
Thsbk you for developing this keyboard.

Glide gestures work on my old honor view 10 (android 10) though suggestions/alternative words are not shown in the smart bar (as they are when typing).

On my xperia 5 II (gestures 11) gestures do not work, unfortunately.

**Comment by VA1DER** [comment]
Glide typing is non-functional in 3.14 (75) on FairPhone 3 and Android 10.

**Comment by patrickgold** [comment]
What is the status of this bug in 0.3.16-beta02? Does it still not work for you or does glide typing (in the current generic Engligh US only config) work now?

**Comment by tsiflimagas** [comment]
Another ping for update of the status here.

**Comment by vacuumbeef** [comment]
It's strange but everything was great for like 1-2 years. Then I was away from android for some time, but when I flashed it back - gestures stopped working for me. Around month ago.

**Comment by lm41** [comment]
@vacuumbeef If you are using Android 14+ this bug may have been resolved by #2959. You can test this by installing the debug apk linked in the PR.

**Comment by vacuumbeef** [comment]
@lm41 it works, finally, you're awesome
